### PR TITLE
Add `digest` file docs to developer docs

### PR DESCRIPTION
**Pull request by palinatolmach**
Closes https://github.com/runtimeverification/gitbook-kontrol/issues/44.

This PR adds `Developer Docs` section to our documentation (I think) and adds the documentation for the `digest` file copied from https://docs.google.com/document/d/18HEqdVx2pi2tRjq-4GvYy80I4RuEgHxwQX2zniHEAfo/edit. @asavienko if there are any further changes you'd like to make in this file, please let me know or do it here. Thanks!

### CODE REVIEW

File: developer-docs/digest-file.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+The digest file consists of the following top-level fields:
+- `kompilation`: A string representing the SHA-256 hash of the compilation artifact.
+- `foundry`: A string representing the SHA-256 hash of the Foundry used to compile the contracts.
+- `methods`: An object containing key-value pairs, where each key represents a function name and an object with the following fields:
+- `method`: A string representing the SHA-256 hash of the method signature.
+- `contract`: A string representing the SHA-256 hash of the contract that contains the method.
+
+### Kompilation
+kompilation digest is computed here. It concatenates the hashes of `contracts.k`, `foundry.k` and any additional `.k` file passed to kontrol build using `--requires` (such as `lemmas.k`); then hashes the concatenated string once more.
```
Comment (anvacaru) [inline developer-docs/digest-file.md:16]: Hyperlinks missing?

Comment (palinatolmach) [reply developer-docs/digest-file.md:16]: There weren't any in the original file here, but that's a good point — I'll add some to the files from our test suite.

**Review by anvacaru — APPROVED** [review]
LGTM! left a single comment.

File: developer-docs/digest-file.md (hunk 1 above)
Comment (palinatolmach) [reply developer-docs/digest-file.md]: Sorry, I understand what you mean now — I added a missing link to the implementation and golden files from the test suite https://github.com/runtimeverification/gitbook-kontrol/pull/45/commits/2b0256eb06d9f997f9e42a9ac06c73bbfafaf441.

**Pull request merged**